1-source-code/2-writing/2-gofmt-and-friends.md

```diff
@@ -302,6 +302,59 @@ type Person struct {
 }
 ```
 
+## `eg`
+
+`eg` is an example-based refactoring tool.
+To use it, you simply create a template file and then provide the paths you wish to refactor.
+`eg` is particulary useful for changing function or API usages.
+
+### Installing `eg`
+
+```bash
+$ go get golang.org/x/tools/cmd/eg
+$ ^get^build
+```
+
+### Writing a template
+
+Here's a simple template (`template.go`) to get you started.
+Templates need to be buildable Go files with `before` and `after` functions.
+
+```go
```

Comment:
please use embedmd for this, see an example of how it works in https://github.com/campoy/go-tooling-workshop/blob/master/1-source-code/1-workspace/1-intro.md

Reply:
Done! This is a great tool :smile_cat: